### PR TITLE
Kubernetes Platform - Implement delete all containers #222

### LINKED ISSUE #222: Kubernetes Platform - Implement delete all containers

### PR DESCRIPTION
**Pull request by enudler**
closes #222

### CODE REVIEW

File: src/jobs/models/kubernetes/jobConnector.js [hunk 1]
```diff
@@ -76,10 +112,10 @@ async function getJobControllerUid(jobPlatformName, platformSpecificInternalRunI
     return controllerUid;
 }
 
-async function getLogsByPodsNames(podsNames) {
+async function getLogsByPodsNames(podsNames, predatorRunnerPrefix) {
```
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]: `predatorRunnerPrefix` isn't used

File: src/jobs/models/kubernetes/jobConnector.js [hunk 2]
```diff
@@ -52,16 +52,52 @@ module.exports.stopRun = async (jobPlatformName, platformSpecificInternalRunId)
     await requestSender.send(options);
 };
 
-module.exports.getLogs = async (jobPlatformName, platformSpecificInternalRunId) => {
+module.exports.getLogs = async (jobPlatformName, platformSpecificInternalRunId, predatorRunnerPrefix) => {
```
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]: `predatorRunnerPrefix` isnt used

File: src/jobs/models/kubernetes/jobConnector.js [hunk 3]
```diff
@@ -52,16 +52,52 @@ module.exports.stopRun = async (jobPlatformName, platformSpecificInternalRunId)
...
     let jobControllerUid = await getJobControllerUid(jobPlatformName, platformSpecificInternalRunId);
 
     let podsNames = await getPodsByLabel(jobControllerUid);
 
-    let logs = await getLogsByPodsNames(podsNames);
+    let logs = await getLogsByPodsNames(podsNames, predatorRunnerPrefix);
```
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]: `predatorRunnerPrefix` isnt used

File: tests/integration-tests/jobs/createJobKubernetes-test.js [hunk 4]
```diff
@@ -370,6 +371,48 @@ describe('Create job specific kubernetes tests', async function () {
 
                         should(getJobsFromService.status).eql(404);
                     });
+
+                    it('Delete the containers', async () => {
```
Comment (NivLipetz) [inline tests/integration-tests/jobs/createJobKubernetes-test.js]:
```suggestion
                    it('Delete only terminated predator containers', async () => {
```

File: tests/unit-tests/jobs/models/docker/jobConnector-test.js [hunk 5]
```diff
@@ -212,4 +212,34 @@ describe('Docker job connector tests', function () {
...
     });
-});
\ No newline at end of file
+
+    describe('Delete all containers', () => {
+        it('Delete 3 containers', async () => {
+            listContainersStub.resolves([
+                { Names: ['predator-runner.0'] },
+                { Names: ['predator-runner.1'] },
+                { Names: ['predator-runner.2'] }]);
```
Comment (NivLipetz) [inline tests/unit-tests/jobs/models/docker/jobConnector-test.js]: add another entry to array resolved with `Names: ['someJob.0']` and expect same results

File: src/jobs/models/jobManager.js [hunk 6]
```diff
@@ -65,17 +66,15 @@ module.exports.stopRun = async function (jobId, runId) {
 
 module.exports.deleteAllContainers = async function () {
     const configData = await configHandler.getConfig();
-    let runnerImage = configData.runner_docker_image;
     const jobConnector = require(`./${configData.job_platform.toLowerCase()}/jobConnector`);
-    let runnerImageWithoutTag = runnerImage.substring(0, runnerImage.indexOf(':'));
-    let result = await jobConnector.deleteAllContainers(runnerImageWithoutTag);
+    let result = await jobConnector.deleteAllContainers(PREDATOR_RUNNER_PREFIX);
```
Comment (NivLipetz) [inline src/jobs/models/jobManager.js]: the function is receiving `PREDATOR_RUNNER_PREFIX` as the argument, but i'd expect it to receive `predator` or `JOB_PLATFORM_NAME`

In my opinion `PREDATOR_RUNNER_PREFIX` isn't really used (also in the other functions) and can be removed

File: src/jobs/models/kubernetes/jobConnector.js [hunk 7]
```diff
@@ -52,16 +52,52 @@ module.exports.stopRun = async (jobPlatformName, platformSpecificInternalRunId)
...
 
     return logs;
 };
 
+module.exports.deleteAllContainers = async (jobPlatformName) => {
```
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]:
```suggestion
module.exports.deleteAllContainers = async (predatorRunnerAppLabel) => {
```

File: src/jobs/models/jobManager.js [hunk 8]
```diff
@@ -8,7 +8,8 @@ const logger = require('../../common/logger'),
     databaseConnector = require('./database/databaseConnector');
 
 let cronJobs = {};
-const JOB_PLATFORM_NAME = 'predator.%s';
+const PREDATOR_RUNNER_PREFIX = 'predator';
```
Comment (NivLipetz) [inline src/jobs/models/jobManager.js]: In my opinion `PREDATOR_RUNNER_PREFIX` isn't really used (also in the other functions) and can be removed

File: src/jobs/models/kubernetes/jobConnector.js [hunk 9]
```diff
@@ -52,16 +52,52 @@ module.exports.stopRun = async (jobPlatformName, platformSpecificInternalRunId)
...
+        const pods = await getPodsByLabel(jobs[i].metadata.uid);
+        allPredatorRunnersPods = allPredatorRunnersPods.concat(pods);
+    }
+
+    let deleted = 0;
+    for (let i = 0; i < allPredatorRunnersPods.length; i++) {
+        let pod = await getPodByName(allPredatorRunnersPods[i]);
+
+        let containers = pod.status.containerStatuses;
+        containers = containers.find(o => o.name === 'predator-runner');
```
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]: `predator-runner` is used twice in this file, use it as a const

`const PREDATOR_RUNNER = 'predator-runner'`

File: src/jobs/models/kubernetes/jobConnector.js [hunk 10]
```diff
@@ -52,16 +52,52 @@ module.exports.stopRun = async (jobPlatformName, platformSpecificInternalRunId)
...
+        if (containers && containers.state.terminated && containers.state.terminated.finishedAt) {
+            await deleteContainer(pod);
+            deleted++;
+        }
+    }
+    return { deleted };
+};
+
+async function getAllPredatorRunnerJobs(jobPlatformName) {
```
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]:
```suggestion
async function getAllJobsByLabel(jobPlatformName) {
```

File: src/jobs/models/kubernetes/jobConnector.js [hunk 11]
```diff
@@ -52,16 +52,52 @@ module.exports.stopRun = async (jobPlatformName, platformSpecificInternalRunId)
...
+    let jobs = await getAllPredatorRunnerJobs(jobPlatformName);
+    let allPredatorRunnersPods = [];
+    for (let i = 0; i < jobs.length; i++) {
+        const pods = await getPodsByLabel(jobs[i].metadata.uid);
+        allPredatorRunnersPods = allPredatorRunnersPods.concat(pods);
+    }
+
+    let deleted = 0;
+    for (let i = 0; i < allPredatorRunnersPods.length; i++) {
+        let pod = await getPodByName(allPredatorRunnersPods[i]);
```
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]:
```suggestion
        const pod = await getPodByName(allPredatorRunnersPods[i]);
```

File: src/jobs/models/kubernetes/jobConnector.js (hunk 9 above)
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]:
```suggestion
        const predatorRunnerContainer = containers.find(o => o.name === 'predator-runner');
```

File: src/jobs/models/kubernetes/jobConnector.js [hunk 12]
```diff
@@ -52,16 +52,52 @@ module.exports.stopRun = async (jobPlatformName, platformSpecificInternalRunId)
...
+
+        let containers = pod.status.containerStatuses;
+        containers = containers.find(o => o.name === 'predator-runner');
+        if (containers && containers.state.terminated && containers.state.terminated.finishedAt) {
```
Comment (NivLipetz) [inline src/jobs/models/kubernetes/jobConnector.js]:
```suggestion
        if (const predatorRunnerContainer && const predatorRunnerContainer.state.terminated && const predatorRunnerContainer.state.terminated.finishedAt) {
```

File: src/jobs/models/jobManager.js (hunk 6 above)
Comment (NivLipetz) [inline src/jobs/models/jobManager.js]:
```suggestion
    let result = await jobConnector.deleteAllContainers(PREDATOR_RUNNER_JOB_APP_LABEL);
```

File: src/jobs/models/jobManager.js (hunk 8 above)
Comment (NivLipetz) [inline src/jobs/models/jobManager.js]: this resembles more the `PREDATOR_RUNNER_JOB_APP_LABEL`

File: src/jobs/models/jobManager.js (hunk 6 above)
Comment (enudler) [reply src/jobs/models/jobManager.js]: the job manager is common for a all 3 paltform
so it's better not use the 'label'

File: src/jobs/models/jobManager.js (hunk 8 above)
Comment (enudler) [reply src/jobs/models/jobManager.js]: true but only when kubernetes is relevant

**Pull request merged**